doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,224 @@
+| `maximumDepth` | `size_t` | Maximum tree depth for weak learner decision tree | 2 |
+
+***Note:*** different types can be used for `data` (e.g.,
+`arma::fmat`, `arma::sp_mat`). 
+
+### Training
+
+If training is not done as part of the constructor call, it can be done with one
+of the following versions of the `Train()` member function:
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical data without using any decision tree arguments.
+
+---
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical data using weak learner model's arguments `minimumLeafSize`, `minimumGainSplit` and `maximumDepth`.
+
+---
+
+Types of each argument are the same as in the table for constructors
+[above](#constructor-parameters).
+
+***Note:***: Training is not incremental.  A second call to `Train()` will retrain the
+model from scratch.
+
+### Classification
+
+Once a `GradBoosting` object is trained, the `Classify()` member function can be used
+to make class predictions for new data.
+
+ * `size_t predictedClass = gb.Classify(point)`
+    - ***(Single-point)***
+    - Classify a single point, returning the predicted class.
+
+---
+
+ * `gb.Classify(point, prediction)`
+    - ***(Single-point)***
+    - Classify a single point and compute class probabilities.
+    - The predicted class is stored in `prediction`.
+
+---
+
+ * `gb.Classify(data, predictedLabels)`
+    - ***(Multi-point)***
+    - Classify a set of points.
+    - The prediction for data point `i` can be accessed with `predictedLabels[i]`.
+
+---
+
+#### Classification Parameters:
+
+| **usage** | **name** | **type** | **description** |
+|-----------|----------|----------|-----------------|
+| _single-point_ | `point` | [`arma::vec`](../matrices.md) | Single point for classification. |
+| _single-point_ | `prediction` | `size_t&` | `size_t` to store class prediction into. |
+||||
+| _multi-point_ | `data` | [`arma::mat`](../matrices.md) | Set of [column-major](../matrices.md#representing-data-in-mlpack) points for classification. |
+| _multi-point_ | `predictedLabels` | [`arma::Row<size_t>&`](../matrices.md) | Vector of `size_t`s to store class prediction into.  Will be set to length `data.n_cols`. |
+
+***Note:*** different types can be used for `data` and `point` (e.g.
+`arma::fmat`, `arma::sp_mat`, `arma::sp_vec`, etc.).  However, the element type
+that is used should be the same type that was used for training.
+
+### Other Functionality
+
+ * `WeakLearner(size_t i)` returns a `DecisionTreeRegressor&` indicating which weak learner 
+    is being used. 
+
+ * `gb.NumWeakLearners()` returns a `size_t` indicating how many weak learners are being
+    used by the model.
+
+ * `gb.NumClasses()` returns a `size_t` indicating the number of classes the
+   model was trained on.
+
+ * `gb.SetNumWeakLearners(size_t x)` sets the number of weak learners to `x` explicitly.
```

Comment:
For consistency with other classes, I would rename this to `gb.NumWeakLearners(const size_t)` in the code, and then document it here with `NumWeakLearners()`:

```
 * `gb.NumWeakLearners()` returns a `size_t` indicating how many weak learners are being
    used by the model.
    - `gb.NumWeakLearners(x)` will set the number of weak learners that will be used the *next time* `Train()` is called to `x`.
```